corehq/apps/reports/static/reports/v2/js/datagrid/data_models.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * todo add docstring
+ */
+
+hqDefine('reports/v2/js/datagrid/data_models', [
+    'jquery',
+    'knockout',
+    'underscore',
+], function (
+    $,
+    ko,
+    _
+) {
+    'use strict';
+
+    var scrollingDataModel = function (endpoint) {
```

Comment:
I will definitely think about this a bit more though. It will definitely be far easier to implement if we do pagination vs. infinite scroll, so maybe I should go back to that for round one and see how strongly the rest of product still feels about infinite scroll...

Reply:
If there's a conversation about infinite scroll sometime, I'd like to join. We ought to have a good rationalization for using / not using it.